subprojects/snapshots/src/main/java/org/gradle/internal/vfs/watch/impl/NoopFileWatcherRegistry.java

```diff
@@ -30,9 +30,11 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(NoopFileWatcherRegistry.class);
 
     private static final FileWatcherRegistry INSTANCE = new NoopFileWatcherRegistry();
+    private static final FileWatchingStatistics NOOP_FILE_WATCHING_STATISTICS = new FileWatchingStatistics(true, 0);
```

Comment:
This will make us log that the VFS was dropped due to an unknown event. I think that's fine for now, but we need something better for when file watching is disabled. Perhaps we should have the `FileWatcherRegistry` issue those logs instead, and just do nothing in this implementation?

Reply:
This is not about file watching being disabled, but about file watching being enabled and not supported on the current OS/Architecture. We should issue a more specific warning in this case when you try to enable it.

Reply:
That's what I wanted to say, too. :) I think moving the logging to the `FileWatcherRegistry` is going to be necessary to issue a more specific message.